---

## VM options
_vm_cleanup: false
_vm_purge: false

vm_start_on_create: "{{ inventory_hostname in groups['provisioned_vms'] }}"
vm_pub_key: "{{ lookup('env', 'HOME') }}/.ssh/id_rsa.pub"
vm_flags: >-
  --graphics vnc,listen=0.0.0.0
  --controller scsi,model=virtio-scsi
  --serial pty
  --parallel pty
  --virt-type kvm
  --memballoon virtio
  --cpu host-passthrough
  --rng /dev/urandom

# Images to pull from for local builds
vm_image_data:
  centos7:
    url: https://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud.qcow2
    path: "{{ vm_image_path | default('/mnt/rhv/images') }}"
    name: CentOS-7-x86_64-GenericCloud.qcow2
    access_user: 'centos'
    cloud_init_packages:
      - NetworkManager
      - qemu-guest-agent
    python_interpreter: "/usr/bin/python"
  fedora28:
    url: https://download.fedoraproject.org/pub/fedora/linux/releases/28/Cloud/x86_64/images/Fedora-Cloud-Base-28-1.1.x86_64.qcow2
    path: "{{ vm_image_path | default('/mnt/rhv/images') }}"
    name: Fedora-Cloud-Base-28-1.1.x86_64.qcow2
    access_user: 'fedora'
    cloud_init_packages:
      - NetworkManager
      - qemu-guest-agent
    python_interpreter: "/usr/bin/python3"
  fedora30:
    url: https://download.fedoraproject.org/pub/fedora/linux/releases/30/Cloud/x86_64/images/Fedora-Cloud-Base-30-1.2.x86_64.qcow2
    path: "{{ vm_image_path | default('/mnt/rhv/images') }}"
    name: Fedora-Cloud-Base-30-1.2.x86_64.qcow2
    access_user: 'fedora'
    cloud_init_packages:
      - NetworkManager
      - qemu-guest-agent
    python_interpreter: "/usr/bin/python3"
  rhel7:
    url: "{{ redhat_7_cloud_image_url | default('undefined') }}"
    path: "{{ vm_image_path | default('/mnt/rhv/images') }}"
    name: rhel-server-7-x86_64-kvm.qcow2
    access_user: 'cloud-user'
    cloud_init_packages:
      - NetworkManager
      - qemu-guest-agent
    python_interpreter: "/usr/bin/python"
  rhel8.0:
    url: "{{ redhat_80_cloud_image_url | default('undefined') }}"
    path: "{{ vm_image_path | default('/mnt/rhv/images') }}"
    name: rhel-8.0-x86_64-kvm.qcow2
    access_user: 'cloud-user'
    cloud_init_packages:
      - NetworkManager
      - qemu-guest-agent
    python_interpreter: "/usr/bin/python3"
  rhel8:
    url: "{{ redhat_8_cloud_image_url | default('undefined') }}"
    path: "{{ vm_image_path | default('/mnt/rhv/images') }}"
    name: rhel-8-x86_64-kvm.qcow2
    access_user: 'cloud-user'
    cloud_init_packages:
      - NetworkManager
      - qemu-guest-agent
    python_interpreter: "/usr/bin/python3"
  rhel8-ppc64le:
    url: "{{ redhat_8_ppc64le_cloud_image_url | default('undefined') }}"
    path: "{{ vm_image_path | default('/mnt/rhv/images') }}"
    name: rhel-8-ppc64le-kvm.qcow2
    access_user: 'cloud-user'
    arch: ppc64le
    cloud_init_packages:
      - NetworkManager
      - qemu-guest-agent
    python_interpreter: "/usr/bin/python3"
  ubuntu1604:
    url: https://cloud-images.ubuntu.com/xenial/current/xenial-server-cloudimg-amd64-disk1.img
    path: "{{ vm_image_path | default('/mnt/rhv/images') }}"
    name: xenial-server-cloudimg-amd64-disk1.img
    access_user: 'ubuntu'
    cloud_init_packages:
      - qemu-guest-agent
    python_interpreter: "/usr/bin/python3"
  ubuntu1804:
    url: https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img
    path: "{{ vm_image_path | default('/mnt/rhv/images') }}"
    name: bionic-server-cloudimg-amd64.img
    access_user: 'ubuntu'
    cloud_init_packages:
      - qemu-guest-agent
    python_interpreter: "/usr/bin/python3"

# Enable image downloads
vm_image_get: true

# Set the vm job user, by default this obtained from the image details
vm_job_user: "{{ vm_image_data[vm_image_name]['access_user'] }}"
